tests/cli/test_model_hook_order.py

```diff
@@ -0,0 +1,107 @@
+from ert_shared.models import (
+    EnsembleSmoother,
+    MultipleDataAssimilation,
+    IteratedEnsembleSmoother,
+)
+import inspect
+
+from unittest.mock import MagicMock
+
+EXPECTED_CALL_ORDER = [
+    "PRE_SIMULATION",
+    "POST_SIMULATION",
+    "PRE_FIRST_UPDATE",
+    "PRE_UPDATE",
+    "POST_UPDATE",
+    "PRE_SIMULATION",
+    "POST_SIMULATION",
+]
+
+
+def test_hook_call_order_ensemble_smoother(monkeypatch):
+    test_class = EnsembleSmoother
+    mock_parent = MagicMock()
+    mock_sim_runner = MagicMock()
+    mock_parent.runWorkflows = mock_sim_runner
+    test_module = inspect.getmodule(test_class)
+    monkeypatch.setattr(test_module, "EnkfSimulationRunner", mock_parent)
+    monkeypatch.setattr(test_module, "ERT", MagicMock())
+    test_class.runSimulations(MagicMock(), MagicMock())
+    assert mock_sim_runner.mock_calls
+    for expected_call, actual_call in zip(
+        EXPECTED_CALL_ORDER, mock_sim_runner.mock_calls
+    ):
+        assert expected_call in str(actual_call)
+
+
+def test_hook_call_order_es_mda(monkeypatch):
+    test_class = MultipleDataAssimilation
+    minimum_args = {
+        "start_iteration": 0,
+        "weights": [1],
+        "analysis_module": "some_module",
+    }
+    mock_sim_runner = MagicMock()
+    mock_parent = MagicMock()
+    mock_parent.runWorkflows = mock_sim_runner
+
+    test_module = inspect.getmodule(test_class)
+    monkeypatch.setattr(test_module, "EnkfSimulationRunner", mock_parent)
+    monkeypatch.setattr(test_module, "ERT", MagicMock())
+
+    test_class = test_class()
+    test_class.create_context = MagicMock()
+    test_class.checkMinimumActiveRealizations = MagicMock()
+    test_class.parseWeights = MagicMock(return_value=[1])
+    test_class.setAnalysisModule = MagicMock()
+    test_class.ert = MagicMock()
+    test_class.ert.return_value.getEnkfSimulationRunner.return_value.runSimpleStep.return_value = (
```

Comment:
Why the `(1)` instead of just `1`? 🤔

Reply:
I guess it is black, but it does not make the line any shorter...

Can we restructure the line into two lines to make them a bit shorter?

Reply:
Would you prefer this:
```
    sim_runner_mock = MagicMock()
    sim_runner_mock.runSimpleStep.return_value = 1
    test_class.ert.return_value.getEnkfSimulationRunner = sim_runner_mock
```

Reply:
Yes, that seems like a necessary evil... 👍